### ISSUE TITLE
`oxlint-disable-next-line` does not work in some cases

### ISSUE THREAD

**Issue by DreierF** [oxc-project/tsgolint#110]
I didn't really understand yet in which cases `oxlint-disable-next-line` does not work correctly, but

```
function foo() {
	// oxlint-disable-next-line no-base-to-string
	return String({});
}
```
for example still reports the error even though it should be silenced.

```
  × typescript-eslint(no-base-to-string): '{}' will use Object's default stringification format ('[object Object]') when stringified.
    ╭─[<file>]
 86 │     // oxlint-disable-next-line no-base-to-string
 87 │     return String({});
    ·                   ──
 88 │ }
```

**Comment by br1anchen** [comment]
Using 
```
    "oxlint": "^1.12.0",
    "oxlint-tsgolint": "^0.0.4",
```

in following case:
```ts
        // oxlint-disable-next-line typescript/no-redundant-type-constituents 
        user_uuid: UserID | 'currentUser'
```
none of the `disable` comment works
tried with `oxlint-disable`, `oxlint-disable-line` and `oxlint-disable-next-line`

**Comment by camc314** [comment]
tracking in https://github.com/oxc-project/oxc/issues/13491
